### ISSUE TITLE
Microsoft flags YaraX compiled binary as Trojan malware

### ISSUE THREAD

**Issue by Benji377** [VirusTotal/yara-x#200]
## Intro
I have a public repo where I build YARA rules using a Python script, the binary is then uploaded to a release and later downloaded by our application. The application uses the binary to scan files.

## Problem
The issue is, that once we download the binary from the release page, Microsoft Windows Defender immediately deletes it and reports it as a Trojan. Has anybody had the same issue?
Here is an extract of how the binary is built:
```py
...
# Iterate through the rules folder and retrieve each yar file in the folder and subfolders
rules_folder = 'rules'
for root, dirs, files in os.walk(rules_folder):
    for file in files:
        if file.endswith('.yar'):
            # Get the content of the file as string
            with open(os.path.join(root, file), 'r') as rule_file:
                content = rule_file.readlines()
                # Add the rule to the compiler
                try:
                    compiler.add_source(''.join(content))
                    number_of_rules += 1
                except Exception as e:
                    error_rules.append(file)
                    print(f'Error compiling {file}: {e}')

# Compile the rules
rules = compiler.build()

# Save the compiled rules into a binary file
with open('rulepirus.yarac', 'wb') as write_file:
    rules.serialize_into(write_file)
...
```

## Links
- [Python builder.py script](https://github.com/Raspirus/yara-rules/blob/main/scripts/builder.py)
- [Yara binary release page](https://github.com/Raspirus/yara-rules/releases/tag/v1.1.0)
- [Yara Rules source](https://yarahq.github.io/)
- [Application page](https://github.com/Raspirus/Raspirus)
- [VirusTotal scan report](https://www.virustotal.com/gui/url/ab6026851d2d18730865a5018107706252992917a7ebdb12eae92616e39cfe7e)

**Comment by GamingGuy003** [comment]
Co Programmer of OP here. This has been happening for multiple yara-x versions including the latest one btw

**Comment by plusvic** [comment]
This may happen because compiled rules include a WebAssembly module in compiled form, and Microsoft Windows Defender may consider that suspicious. I'll investigate.

**Comment by plusvic** [comment]
However, I strongly suggest you reconsidering the distribution of YARA rules in binary form to your users. Using compiled YARA rules from third-parties is highly discouraged because it constitutes a security risk. Bear in mind that compiled YARA rules contain executable code, therefore they can be manipulated to execute malicious payloads. 

Also, compiled YARA rules at this moment are guaranteed to run in the same platform that was used for compiling. You if you compile them in 64-bits Linux, you must use them in 64-bits Linux, they won't work on Windows or macOS. You will also encounter issues when compiling the rules with some version of YARA and trying o use it with some other version.

The bottom line is that compiled rules are a mechanism for avoiding the re-compilation of large sets of rules, but assuming that you are compiling the rules yourself and that you are not using it in cross-platform environment. They are not intended to be used as a mechanism for distributing closed-source YARA rules.

**Comment by Benji377** [comment]
Hey @plusvic, thank you for your fast response.

I wasn't aware of this restriction. I did notice that we had to recompile the rules for each yara-x update but I didn't know it also depends on the OS.

Thank you so much for pointing this out. So it is probably better to download all the yara rules and compile them on each user's device separately.
We are planning to cross-compile the application to ARM too, so this is quite important to know.
